### DISCUSSION TITLE
Possibly creating feature vectors using full text of records

### DISCUSSION THREAD

**Discussion by rohitgarud** [asreview/asreview#1455] [Ideas]
Hi all,
There is a [longformer model by AllenAI](https://huggingface.co/allenai/led-base-16384) with a context length of 16384, which was used in the [QASPER paper](https://arxiv.org/abs/2105.03011) by researchers for embedding full texts for Question-Answering task can be used as feature extractor using the SBERT feature extraction model with the recently updated functionality (Python API) to possibly obtain feature vectors (embeddings) for the full texts of the papers in the dataset. This can be one of the solutions for the use cases where the required information is somewhere in the full text and is not available in the abstract.
Please give me your feedback. Thank you.

**Comment by J535D165** [comment]
This looks very promising. 

In OpenAlex are ngrams available, I'm wondering if we can use those somehow in this feature extraction workflow as well. Based on your idea.

**Reply by rohitgarud** [reply]
Yes, @J535D165 .. we can use those ngrams to create feature vectors by combining them into a single text. We can also extract full text from PDFs.. this will need some extra work but we can get the entire text. This can also be used to extract only the methodology section or some other sections from the text and use it for screening instead of abstracts (many users are having this use case https://github.com/asreview/asreview/discussions/1398 https://github.com/asreview/asreview/discussions/1384)
